dvc/proc/manager.py

```diff
@@ -63,17 +65,62 @@ def spawn(self, args: Union[str, List[str]], name: Optional[str] = None):
             pid,
         )
 
-    def send_signal(self, name: str, signal: int):
-        """Send `signal` to the specified named process."""
-        raise NotImplementedError
-
-    def kill(self, name: str):
-        """Kill the specified named process."""
-        raise NotImplementedError
-
-    def terminate(self, name: str):
-        """Terminate the specified named process."""
-        raise NotImplementedError
+    if sys.platform == "win32":
+
+        def send_signal(self, name: str, sig: int):
+            """Send `signal` to the specified named process."""
+            # pylint: disable=no-member
+            process_info = self[name]
+            if process_info.returncode is None:
+                return
+            if sig == signal.SIGTERM:
+                self.terminate(name)
+            elif sig == signal.CTRL_C_EVENT:
+                os.kill(process_info.pid, signal.CTRL_C_EVENT)
+            elif sig == signal.CTRL_BREAK_EVENT:
+                os.kill(process_info.pid, signal.CTRL_BREAK_EVENT)
+            else:
+                raise ValueError("Unsupported signal: {}".format(sig))
+
+    else:
+
+        def send_signal(self, name: str, sig: int):
+            """Send `signal` to the specified named process."""
+            process_info = self[name]
+            if process_info.returncode is None:
+                os.kill(process_info.pid, sig)
+
+    if sys.platform == "win32":
+
+        def terminate(self, name: str):
+            """Terminate the specified named process."""
+            import _winapi  # pylint: disable=import-error
+
+            process_info = self[name]
+            if process_info.returncode is not None:
+                return
+            PROCESS_TERMINATE = 1
+            handle = _winapi.OpenProcess(
+                PROCESS_TERMINATE, False, process_info.pid
+            )
+            try:
+                _winapi.TerminateProcess(handle, 1)
+            except PermissionError:
+                rc = _winapi.GetExitCodeProcess(handle)
+                if rc == _winapi.STILL_ACTIVE:
+                    raise
+                process_info.returncode = rc
```

Comment:
One other suggestion, it may be better to just define this as `_terminate_nt(self, ...)`, and then call it from `send_signal()` for `SIGKILL`/`SIGTERM`. That way, we could keep all of the platform specific handling in `send_signal()`, and not need platform specific handling in `kill()`/`terminate()` at all (since they just call `self.send_signal()`.

Reply:
Also, this could probably be broken up into a separate helper function for getting the windows process handle, which may be something we end up needing in the future.

Reply:
Actually looking at the python [docs](https://docs.python.org/3/library/os.html), I think we can probably get rid of all of this code entirely and just call `os.kill(pid, -1)` instead of trying to call the windows `TerminateProcess()` ourselves

> Any other value for sig will cause the process to be unconditionally killed by the TerminateProcess API, and the exit code will be set to sig 

@karajan1001

Reply:
Yes, but still need to handle these `SIGKILL` and `CTRL_C_EVENT` because they might not exist in some of the platforms

Reply:
Right, I meant for the kill/terminate case on windows we can just call `os.kill(pid, -1)`, we don't need to take care of using windows process handles + `TerminateProcess()` ourselves